code/clue.js

```diff
@@ -1,59 +1,227 @@
+
+
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+return array[Math.floor(Math.random() * array.length)]
 }
+//console.log(randomSelector(suspects));
+
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
+const mystery = {
+  killer : "",
+  weapons : "",
+  room : ""
+};
+
 // With a killer, a weapon and a room.
 // The values will be set later.
 
 // This function will be invoked when you click on the killer card.
+
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
+  
+  //This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerImage').src = mystery.killer.image;
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+}
+
+const pickWeapon = () => {
+  // This will randomly select a weapon. And add that to the mystery object.
+  mystery.weapons = randomSelector(weapons)
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('weaponCard').style.background = mystery.weapons.color
+  document.getElementById('weaponName').innerHTML = `${mystery.weapons.name} <br> Weight:${mystery.weapons.weight}`
+}
+const pickRoom = () => {
+  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  mystery.room = randomSelector(rooms)
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('roomCard').style.background = mystery.room.color
+  document.getElementById('roomName').innerHTML = `${mystery.room}`
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+const theMystery = document.getElementById('mystery');
+const revealButton = document.getElementById (`revealButton`);
+const mysteryAnswer = document.getElementById(`mystery`)
+
+const revealMystery = () => {
+  if (mystery.killer.firstName === undefined
+    || mystery.weapons.name === undefined
+    || mystery.room === undefined) 
```

Comment:
This logic allows you to click the killer and weapon card and press reveal without clicking the room card.  

This is because of line 169.

Consider the following idea
```Javascript
const mystery = {
  killer: undefined,
  weapons: undefined,
  room: undefined
}
```

Try to understand why this would make your page behave differently 👍

Reply:
I tried to follow your example but could not get it to work, the text would not show. 
Instead, I added " " in the if statement too, and that worked, I didn't really understand how that worked and not the undefined option but will figure it out now.